tests/components/recorder/test_websocket_api.py

```diff
@@ -794,6 +794,217 @@ def next_id():
+        {
+            "type": "recorder/statistic_during_period",
+            "statistic_id": statId,
+            "types": ["change"],
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    assert response["result"] == {
+        "change": imported_stats_5min[-1]["sum"] - imported_stats_hours[0]["sum"],
+    }
+
+    async def assert_change_during_fixed(client, start_time, end_time, expected):
+        json = {
+            "type": "recorder/statistic_during_period",
+            "types": ["change"],
+            "statistic_id": statId,
+            "fixed_period": {},
+        }
+        if start_time:
+            json["fixed_period"]["start_time"] = start_time.isoformat()
+        if end_time:
+            json["fixed_period"]["end_time"] = end_time.isoformat()
+
+        await client.send_json_auto_id(json)
+        response = await client.receive_json()
+        assert response["success"]
+        assert response["result"] == {
+            "change": expected,
+        }
+
+    # One hours worth of growth in LTS-only
+    start_time = start.replace(hour=1)
+    end_time = start.replace(hour=2)
+    expect = 60
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # Five minutes of growth in STS-only
+    start_time = start.replace(hour=6, minute=15)
+    end_time = start.replace(hour=6, minute=20)
+    expect = 5
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # Six minutes of growth in STS-only
+    start_time = start.replace(hour=6, minute=14)
+    end_time = start.replace(hour=6, minute=20)
+    expect = 5
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # Six minutes of growth in STS-only
+    # 5-minute Change includes start times exactly on or before a statistics start, but end times are not counted unless they are greater than start.
+    start_time = start.replace(hour=6, minute=15)
+    end_time = start.replace(hour=6, minute=21)
+    expect = 10
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # Five minutes of growth in overlapping LTS+STS
+    start_time = start.replace(hour=5, minute=15)
+    end_time = start.replace(hour=5, minute=20)
+    expect = 5
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # Five minutes of growth in overlapping LTS+STS (start of hour)
+    start_time = start.replace(hour=5, minute=0)
+    end_time = start.replace(hour=5, minute=5)
+    expect = 5
+    await assert_change_during_fixed(client, start_time, end_time, expect)
+
+    # Five minutes of growth in overlapping LTS+STS (end of hour)
+    start_time = start.replace(hour=4, minute=55)
+    end_time = start.replace(hour=5, minute=0)
+    expect = 5
+    # FIXME: this returns None, I might have expected 5 given the behavior of other 5 minute intervals
+    # await assert_change_during_fixed(client, start_time, end_time, expect)
```

Comment:
This test case is supposed to only get data from the tail, however there's a missing case. This makes the test pass:
```diff
(.venv) erik@Thinkstation:~/development/home-assistant_fork$ git diff
diff --git a/homeassistant/components/recorder/statistics.py b/homeassistant/components/recorder/statistics.py
index 6167d67cc85..7364149bebc 100644
--- a/homeassistant/components/recorder/statistics.py
+++ b/homeassistant/components/recorder/statistics.py
@@ -1495,6 +1495,10 @@ def statistic_during_period(
                 else end_time.replace(minute=0, second=0, microsecond=0)
             )
             tail_end_time = end_time
+        elif tail_only:
+            tail_start_time = start_time
+            tail_end_time = end_time
+

         # Calculate the main period
         main_start_time: datetime | None = None
```

I'm not sure if this is entirely correct, or if the existing conditions should be modified.